### DISCUSSION TITLE
Search gives too many results

### DISCUSSION THREAD

**Discussion by 1Cedric1** [osmandapp/OsmAnd#14103] [Ideas]
Hello,
Wanting to drive to Chatel, a village in France, I am typing "Chatel" in the search field, which returns more than 200 results. "Rue de Chatel", "Chatel Saint Denis", etc... not alphabetically sorted are showned, and we have to scroll a long time to eventually find the village of "Chatel".
Is there an option to show only the results for an exact search, to find the destination faster? If yes, how? and if not, would it be possible to implement it?
Thank you for your interest and looking forward to read you.
Kind regards.

**Comment by Lee-Carre** [comment]
Well, it's not really about the _count_ of returned matches, but the _ordering (prioritisation)_ of matches.
Ideally, only a single match (the correct / intended one) is needed; but how would software know which one that is(?) 😄.

Off the top of my head, it does seem sensible to prioritise exact matches (followed by almost-exact matches, to allow for minor differences, such variations involving [blank-space](/osmandapp/OsmAnd/issues/13783) and non-space separator-characters).
One issue might be unintended matching of something irrelevant when inputting a substring. Though, I suppose, if users are informed that exact matches are prioritised, then they'd know to input the complete name (or use their favourite auto-complete / suggestions tool).

One problem for finding the right match, is that all different types of features must be queried. I wonder if it might be feasible to give OsmAnd clues (via a filter, for example) which type of matches it should include / exclude (e.g. only roads, only place names, anything except shops, towns but not cities, …). Else, possibly dynamic ordering based on detected context (but I've no good ideas for how that would work).

Another problem is; how far (from current position) should be searched?
* sometimes you'll want the nearest match (e.g. shops)
* sometimes, you're planning a longer trip, and want to find a named road in a different city

Search is a tricky problem, in computer science.

Edit: a different possibility is to prioritise matches within the viewport of the map (or nearest to its centroid). This mechanic would need to be communicated to users, somehow.
See also #14049.

**Reply by 1Cedric1** [reply]
Hello, and thank you very much for your answer.

Just to illustrate my question / remark, as I would like to travel to Berlin e.g.

Search with Google Maps: the first hit is what I search (= the city of Berlin)

 
![image](https://user-images.githubusercontent.com/61581150/164878206-31da57ef-3096-4589-8098-0251e0115b37.png)

When I do the same search on Osmand, many many hits are shown and I have to scroll many times to find the city of Berlin, what's not ideal.

![image](https://user-images.githubusercontent.com/61581150/164878273-af4f8453-f5a8-450c-92ae-6ae6644da388.png)

An idea could be to select first if I would like to search a city or a street, what could be a good filter. So just an proposal, if you find it useful.

**Comment by vshcherb** [comment]
This release 4.2 we've prioritized exact searches much higher, so it should be fixed

**Reply by 1Cedric1** [reply]
hello, and thank you for your response. Looking forward to using it!

**Comment by ThomasMeister123** [comment]
Please provide a hierachical search: first search for cities. If you type something, there ore just citys in the results, no streets/POIs/whatever. Then you select a city an then you type a street. And now there aro (only!) streets in the selected city in the results (no POIS!)

**Reply by tommycrock** [reply]
Is this not what you get with the address search and 'first specify city/town/locality' option?

**Reply by ThomasMeister123** [reply]
OK. Yes, that works nearly like what I'm looking for. Thanx
